app/tests/test_views.py

```diff
@@ -0,0 +1,77 @@
+from flask.ext.testing import TestCase
+
+from app.factory import create_app, user_datastore
+from app.models import User
+
+from .test_models import DbTestCase
+
+LOGGED_IN_SENTINEL = '<a href="/me"'
+
+class ViewTestCase(DbTestCase):
+    def create_app(self):
+        config = self.BASE_APP_CONFIG.copy()
+        config.update(
+            WTF_CSRF_ENABLED=False,
+            SECURITY_REGISTERABLE=True,
+            SECURITY_SEND_REGISTER_EMAIL=False,
+            CACHE_NO_NULL_WARNING=True,
+            SECRET_KEY='test',
+            S3_BUCKET_NAME='test_bucket'
+        )
+        return create_app(config=config)
+
+    def register_and_login(self, username, password):
+        res = self.client.post('/register', data=dict(
+            next='/',
+            email=username,
+            password=password,
+            password_confirm=password,
+            submit='Register'
+        ), follow_redirects=True)
+        self.assert200(res)
+        assert LOGGED_IN_SENTINEL in res.data
+        return res
+
+    def logout(self):
+        res = self.client.get('/logout', follow_redirects=True)
+        self.assert200(res)
+        assert LOGGED_IN_SENTINEL not in res.data
+        return res
+
+    def login(self, username, password):
+        res = self.client.post('/login', data=dict(
+            next='/',
+            submit="Login",
+            email=username,
+            password=password
+        ), follow_redirects=True)
+        self.assert200(res)
```

Comment:
Good point, though I also think that in *most* cases it's expected that the operation is successful, and we want to make sure that it is--what do you think of having an `assert_success=True` default arg for these methods, so that in the rare case that a test doesn't expect the login to succeed, it can change the default behavior?